EventSourceProxy/TypeImplementer.cs

```diff
@@ -744,6 +744,16 @@ private void EmitCallWriteEvent(InvocationContext invocationContext, MethodBuild
 							sourceType = sourceType.GetElementType();
 						mIL.Emit(OpCodes.Box, sourceType);
 					}
+					else if (parameter.CleanTargetType == typeof(string))
+					{
+						// Convert to empty String if the String is null.

```

Comment:
I'm not familiar with IL coding. I see the type test above but I don't see what is doing the null test.

I wish there was a clever solution for any null reference type leading to pass a default instance instead, but reference types may also not have an argument less default constructor (like String doesn't).

Reply:
The check is done on line 752. If not null it will jump to label EndIf. Otherwise it will set the string to empty. 
There is also a test above...